WARN  2016-12-21 19:59:51,436 (StunServiceImpl.java:420) received no response from STUN server stunserver.org/69.172.201.153:3478 at local address /192.168.2.108:0
WARN  2016-12-21 20:00:00,869 (Socks5Transport.java:544) [SOCKS5Transport] could not establish a connection to eazycode@saros-con.imp.fu-berlin.de/Saros, remote side could not connect to any offered stream hosts: item-not-found(404) Could not establish socket with any provided host
ERROR 2016-12-21 20:00:02,313 (ByteStreamTransport.java:105) could not establish bytestream connection to eazycode@saros-con.imp.fu-berlin.de/Saros
Could not establish socket with any provided host: item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.cancelRequest(Socks5BytestreamRequest.java:277)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.accept(Socks5BytestreamRequest.java:247)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.accept(Socks5BytestreamRequest.java:34)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.handleResponse(Socks5Transport.java:315)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.acceptRequest(Socks5Transport.java:471)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport$1.incomingBytestreamRequest(ByteStreamTransport.java:77)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.processRequest(InitiationListener.java:91)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.access$000(InitiationListener.java:37)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener$1.run(InitiationListener.java:63)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:00:30,883 (DataTransferManager.java:415) failed to connect to eazycode@saros-con.imp.fu-berlin.de/Saros using transport: XMPP-Socks5-Transport
java.io.IOException: [SOCKS5Transport] waiting for a response session timed out (30000ms) and could not establish a connection from this side, too:
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishBinaryChannel(Socks5Transport.java:592)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport.connect(ByteStreamTransport.java:48)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.connectInternal(DataTransferManager.java:412)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.connect(DataTransferManager.java:297)
	at de.fu_berlin.inf.dpp.negotiation.IncomingSessionNegotiation.accept(IncomingSessionNegotiation.java:167)
	at de.fu_berlin.inf.dpp.ui.wizards.JoinSessionWizard$2.run(JoinSessionWizard.java:128)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smack.util.SyncPacketSend.getReply(SyncPacketSend.java:54)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:481)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:83)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishBinaryChannel(Socks5Transport.java:512)
	... 6 more
WARN  2016-12-21 20:00:31,973 (SubscriptionHandler.java:148) received error presence package from saros_765913088@conference.saros-con.imp.fu-berlin.de/xzer@saros-con.imp.fu-berlin.de/Saros, condition: not-authorized, message: A password is required to enter this room
WARN  2016-12-21 20:00:46,271 (Socks5Transport.java:411) [SOCKS5Transport] Couldn't accept request but still trying to establish a response connection: Could not establish socket with any provided host
ERROR 2016-12-21 20:00:46,773 (Socks5Transport.java:440) An error occurred while establishing a response connection 
item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smack.util.SyncPacketSend.getReply(SyncPacketSend.java:54)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:481)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:83)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishResponseSession(Socks5Transport.java:631)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport$1.call(Socks5Transport.java:108)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport$1.call(Socks5Transport.java:1)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
ERROR 2016-12-21 20:00:46,794 (ByteStreamTransport.java:105) could not establish bytestream connection to eazycode@saros-con.imp.fu-berlin.de/Saros
java.io.IOException: [SOCKS5Transport] Neither connection could be established. 
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.acceptNewRequest(Socks5Transport.java:446)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.acceptRequest(Socks5Transport.java:474)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport$1.incomingBytestreamRequest(ByteStreamTransport.java:77)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.processRequest(InitiationListener.java:91)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.access$000(InitiationListener.java:37)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener$1.run(InitiationListener.java:63)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
ERROR 2016-12-21 20:13:42,555 (DataTransferManager.java:270) failed to send TransferDescription [elementName=pncl, namespace=de.fu_berlin.inf.dpp, recipient=eazycode@saros-con.imp.fu-berlin.de/Saros, sender=xzer@saros-con.imp.fu-berlin.de/Saros, compress=true], connection=[mode=IBB, id=saros-main-session] eazycode@saros-con.imp.fu-berlin.de/Saros:Error while closing stream: item-not-found(404)
java.io.IOException: Error while closing stream: item-not-found(404)
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession.closeByLocal(InBandBytestreamSession.java:213)
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession.close(InBandBytestreamSession.java:154)
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession$IQIBBOutputStream.writeToXML(InBandBytestreamSession.java:768)
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession$IBBOutputStream.flushBuffer(InBandBytestreamSession.java:707)
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession$IBBOutputStream.flush(InBandBytestreamSession.java:691)
	at java.io.BufferedOutputStream.flush(Unknown Source)
	at java.io.DataOutputStream.flush(Unknown Source)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.send(BinaryChannelConnection.java:315)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendInternal(DataTransferManager.java:268)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendData(DataTransferManager.java:216)
	at de.fu_berlin.inf.dpp.net.internal.XMPPTransmitter.send(XMPPTransmitter.java:90)
	at de.fu_berlin.inf.dpp.negotiation.ProjectNegotiation.notifyCancellation(ProjectNegotiation.java:145)
	at de.fu_berlin.inf.dpp.negotiation.CancelableProcess.terminateProcess(CancelableProcess.java:348)
	at de.fu_berlin.inf.dpp.negotiation.IncomingProjectNegotiation.run(IncomingProjectNegotiation.java:270)
	at de.fu_berlin.inf.dpp.ui.wizards.AddProjectToSessionWizard$5.run(AddProjectToSessionWizard.java:307)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
WARN  2016-12-21 20:13:42,642 (SarosSession.java:655) failed to notify user eazycode@saros-con.imp.fu-berlin.de/Saros about local session stop
java.io.IOException: not connected to eazycode@saros-con.imp.fu-berlin.de/Saros [connection identifier=saros-main-session]
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendData(DataTransferManager.java:206)
	at de.fu_berlin.inf.dpp.net.internal.XMPPTransmitter.send(XMPPTransmitter.java:90)
	at de.fu_berlin.inf.dpp.project.internal.SarosSession.stop(SarosSession.java:651)
	at de.fu_berlin.inf.dpp.project.SarosSessionManager.stopSarosSession(SarosSessionManager.java:340)
	at de.fu_berlin.inf.dpp.negotiation.IncomingProjectNegotiation.executeCancellation(IncomingProjectNegotiation.java:508)
	at de.fu_berlin.inf.dpp.negotiation.CancelableProcess.terminateProcess(CancelableProcess.java:350)
	at de.fu_berlin.inf.dpp.negotiation.IncomingProjectNegotiation.run(IncomingProjectNegotiation.java:270)
	at de.fu_berlin.inf.dpp.ui.wizards.AddProjectToSessionWizard$5.run(AddProjectToSessionWizard.java:307)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
WARN  2016-12-21 20:14:09,411 (Socks5Transport.java:544) [SOCKS5Transport] could not establish a connection to eazycode@saros-con.imp.fu-berlin.de/Saros, remote side could not connect to any offered stream hosts: item-not-found(404) Could not establish socket with any provided host
ERROR 2016-12-21 20:14:20,892 (ByteStreamTransport.java:105) could not establish bytestream connection to eazycode@saros-con.imp.fu-berlin.de/Saros
Could not establish socket with any provided host: item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.cancelRequest(Socks5BytestreamRequest.java:277)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.accept(Socks5BytestreamRequest.java:247)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.accept(Socks5BytestreamRequest.java:34)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.handleResponse(Socks5Transport.java:315)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.acceptRequest(Socks5Transport.java:471)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport$1.incomingBytestreamRequest(ByteStreamTransport.java:77)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.processRequest(InitiationListener.java:91)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.access$000(InitiationListener.java:37)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener$1.run(InitiationListener.java:63)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:14:39,424 (DataTransferManager.java:415) failed to connect to eazycode@saros-con.imp.fu-berlin.de/Saros using transport: XMPP-Socks5-Transport
java.io.IOException: [SOCKS5Transport] waiting for a response session timed out (30000ms) and could not establish a connection from this side, too:
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishBinaryChannel(Socks5Transport.java:592)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport.connect(ByteStreamTransport.java:48)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.connectInternal(DataTransferManager.java:412)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.connect(DataTransferManager.java:297)
	at de.fu_berlin.inf.dpp.negotiation.IncomingSessionNegotiation.accept(IncomingSessionNegotiation.java:167)
	at de.fu_berlin.inf.dpp.ui.wizards.JoinSessionWizard$2.run(JoinSessionWizard.java:128)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smack.util.SyncPacketSend.getReply(SyncPacketSend.java:54)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:481)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:83)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishBinaryChannel(Socks5Transport.java:512)
	... 6 more
WARN  2016-12-21 20:14:39,736 (SubscriptionHandler.java:148) received error presence package from saros_1062711000@conference.saros-con.imp.fu-berlin.de/xzer@saros-con.imp.fu-berlin.de/Saros, condition: not-authorized, message: A password is required to enter this room
WARN  2016-12-21 20:14:55,628 (Socks5Transport.java:411) [SOCKS5Transport] Couldn't accept request but still trying to establish a response connection: Could not establish socket with any provided host
ERROR 2016-12-21 20:14:55,635 (Socks5Transport.java:440) An error occurred while establishing a response connection 
item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smack.util.SyncPacketSend.getReply(SyncPacketSend.java:54)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:481)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:83)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishResponseSession(Socks5Transport.java:631)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport$1.call(Socks5Transport.java:108)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport$1.call(Socks5Transport.java:1)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
ERROR 2016-12-21 20:14:55,642 (ByteStreamTransport.java:105) could not establish bytestream connection to eazycode@saros-con.imp.fu-berlin.de/Saros
java.io.IOException: [SOCKS5Transport] Neither connection could be established. 
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.acceptNewRequest(Socks5Transport.java:446)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.acceptRequest(Socks5Transport.java:474)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport$1.incomingBytestreamRequest(ByteStreamTransport.java:77)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.processRequest(InitiationListener.java:91)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.access$000(InitiationListener.java:37)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener$1.run(InitiationListener.java:63)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:15:28,053 (DataTransferManager.java:154) existing connection [pool id=default:in:eazycode@saros-con.imp.fu-berlin.de/Saros] [mode=IBB, id=default] eazycode@saros-con.imp.fu-berlin.de/Saros was replaced with connection [mode=IBB, id=default] eazycode@saros-con.imp.fu-berlin.de/Saros
ERROR 2016-12-21 20:15:28,073 (BinaryChannelConnection.java:115) [mode=IBB, id=default] eazycode@saros-con.imp.fu-berlin.de/Saros network error: Stream is closed
java.io.IOException: Stream is closed
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession$IBBInputStream.checkClosed(InBandBytestreamSession.java:395)
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession$IBBInputStream.read(InBandBytestreamSession.java:305)
	at java.io.BufferedInputStream.fill(Unknown Source)
	at java.io.BufferedInputStream.read(Unknown Source)
	at java.io.DataInputStream.readUnsignedByte(Unknown Source)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.readNextXMPPExtension(BinaryChannelConnection.java:347)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.access$2(BinaryChannelConnection.java:343)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection$ReceiverThread.run(BinaryChannelConnection.java:106)
ERROR 2016-12-21 20:15:28,239 (BinaryChannelConnection.java:176) failed to gracefully close connection [mode=IBB, id=default] eazycode@saros-con.imp.fu-berlin.de/Saros
java.io.IOException: Error while closing stream: item-not-found(404)
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession.closeByLocal(InBandBytestreamSession.java:213)
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession.close(InBandBytestreamSession.java:153)
	at de.fu_berlin.inf.dpp.net.internal.XMPPByteStreamAdapter.close(XMPPByteStreamAdapter.java:32)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.close(BinaryChannelConnection.java:174)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection$ReceiverThread.run(BinaryChannelConnection.java:119)
WARN  2016-12-21 20:15:59,172 (CancelProjectSharingHandler.java:74) Inv[unknown user]: Received invitation cancel message
ERROR 2016-12-21 20:19:37,631 (ActivitySequencer.java:504) failed to sent activities: [StartFollowingActivity(xzer@saros-con.imp.fu-berlin.de/Saros > eazycode@saros-con.imp.fu-berlin.de/Saros)]
java.io.IOException: not connected to eazycode@saros-con.imp.fu-berlin.de/Saros [connection identifier=saros-main-session]
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendData(DataTransferManager.java:206)
	at de.fu_berlin.inf.dpp.net.internal.XMPPTransmitter.send(XMPPTransmitter.java:90)
	at de.fu_berlin.inf.dpp.project.internal.ActivitySequencer.sendActivities(ActivitySequencer.java:501)
	at de.fu_berlin.inf.dpp.project.internal.ActivitySequencer.access$3(ActivitySequencer.java:482)
	at de.fu_berlin.inf.dpp.project.internal.ActivitySequencer$2.run(ActivitySequencer.java:152)
	at de.fu_berlin.inf.dpp.util.ThreadUtils$1.run(ThreadUtils.java:38)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:19:37,692 (SarosSession.java:655) failed to notify user eazycode@saros-con.imp.fu-berlin.de/Saros about local session stop
java.io.IOException: not connected to eazycode@saros-con.imp.fu-berlin.de/Saros [connection identifier=saros-main-session]
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendData(DataTransferManager.java:206)
	at de.fu_berlin.inf.dpp.net.internal.XMPPTransmitter.send(XMPPTransmitter.java:90)
	at de.fu_berlin.inf.dpp.project.internal.SarosSession.stop(SarosSession.java:651)
	at de.fu_berlin.inf.dpp.project.SarosSessionManager.stopSarosSession(SarosSessionManager.java:340)
	at de.fu_berlin.inf.dpp.project.internal.timeout.SessionTimeoutHandler$3.run(SessionTimeoutHandler.java:115)
	at de.fu_berlin.inf.dpp.util.ThreadUtils$1.run(ThreadUtils.java:38)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:19:37,770 (SubscriptionHandler.java:148) received error presence package from saros_1062711000@conference.saros-con.imp.fu-berlin.de/xzer@saros-con.imp.fu-berlin.de/Saros, condition: item-not-found, message: Conference room does not exist
WARN  2016-12-21 20:33:29,127 (Socks5Transport.java:544) [SOCKS5Transport] could not establish a connection to eazycode@saros-con.imp.fu-berlin.de/Saros, remote side could not connect to any offered stream hosts: item-not-found(404) Could not establish socket with any provided host
ERROR 2016-12-21 20:33:30,347 (ByteStreamTransport.java:105) could not establish bytestream connection to eazycode@saros-con.imp.fu-berlin.de/Saros
Could not establish socket with any provided host: item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.cancelRequest(Socks5BytestreamRequest.java:277)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.accept(Socks5BytestreamRequest.java:247)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.accept(Socks5BytestreamRequest.java:34)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.handleResponse(Socks5Transport.java:315)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.acceptRequest(Socks5Transport.java:471)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport$1.incomingBytestreamRequest(ByteStreamTransport.java:77)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.processRequest(InitiationListener.java:91)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.access$000(InitiationListener.java:37)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener$1.run(InitiationListener.java:63)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:33:59,145 (DataTransferManager.java:415) failed to connect to eazycode@saros-con.imp.fu-berlin.de/Saros using transport: XMPP-Socks5-Transport
java.io.IOException: [SOCKS5Transport] waiting for a response session timed out (30000ms) and could not establish a connection from this side, too:
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishBinaryChannel(Socks5Transport.java:592)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport.connect(ByteStreamTransport.java:48)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.connectInternal(DataTransferManager.java:412)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.connect(DataTransferManager.java:297)
	at de.fu_berlin.inf.dpp.negotiation.IncomingSessionNegotiation.accept(IncomingSessionNegotiation.java:167)
	at de.fu_berlin.inf.dpp.ui.wizards.JoinSessionWizard$2.run(JoinSessionWizard.java:128)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smack.util.SyncPacketSend.getReply(SyncPacketSend.java:54)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:481)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:83)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishBinaryChannel(Socks5Transport.java:512)
	... 6 more
WARN  2016-12-21 20:33:59,352 (SubscriptionHandler.java:148) received error presence package from saros_539475858@conference.saros-con.imp.fu-berlin.de/xzer@saros-con.imp.fu-berlin.de/Saros, condition: not-authorized, message: A password is required to enter this room
ERROR 2016-12-21 20:34:03,425 (CancelableProcess.java:411) cancelled process ISN [remote side: eazycode@saros-con.imp.fu-berlin.de/Saros] because the remote side encountered an error: could not synchronize user list, following users did not respond: xzer@saros-con.imp.fu-berlin.de/Saros
WARN  2016-12-21 20:34:03,467 (SarosSession.java:655) failed to notify user eazycode@saros-con.imp.fu-berlin.de/Saros about local session stop
java.io.IOException: not connected to eazycode@saros-con.imp.fu-berlin.de/Saros [connection identifier=saros-main-session]
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendData(DataTransferManager.java:206)
	at de.fu_berlin.inf.dpp.net.internal.XMPPTransmitter.send(XMPPTransmitter.java:90)
	at de.fu_berlin.inf.dpp.project.internal.SarosSession.stop(SarosSession.java:651)
	at de.fu_berlin.inf.dpp.project.SarosSessionManager.stopSarosSession(SarosSessionManager.java:340)
	at de.fu_berlin.inf.dpp.negotiation.IncomingSessionNegotiation.executeCancellation(IncomingSessionNegotiation.java:92)
	at de.fu_berlin.inf.dpp.negotiation.CancelableProcess.terminateProcess(CancelableProcess.java:350)
	at de.fu_berlin.inf.dpp.negotiation.IncomingSessionNegotiation.accept(IncomingSessionNegotiation.java:181)
	at de.fu_berlin.inf.dpp.ui.wizards.JoinSessionWizard$2.run(JoinSessionWizard.java:128)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
WARN  2016-12-21 20:34:03,571 (SubscriptionHandler.java:148) received error presence package from saros_539475858@conference.saros-con.imp.fu-berlin.de/xzer@saros-con.imp.fu-berlin.de/Saros, condition: item-not-found, message: Conference room does not exist
WARN  2016-12-21 20:34:42,463 (Socks5Transport.java:544) [SOCKS5Transport] could not establish a connection to eazycode@saros-con.imp.fu-berlin.de/Saros, remote side could not connect to any offered stream hosts: item-not-found(404) Could not establish socket with any provided host
ERROR 2016-12-21 20:34:43,398 (ByteStreamTransport.java:105) could not establish bytestream connection to eazycode@saros-con.imp.fu-berlin.de/Saros
Could not establish socket with any provided host: item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.cancelRequest(Socks5BytestreamRequest.java:277)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.accept(Socks5BytestreamRequest.java:247)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.accept(Socks5BytestreamRequest.java:34)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.handleResponse(Socks5Transport.java:315)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.acceptRequest(Socks5Transport.java:471)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport$1.incomingBytestreamRequest(ByteStreamTransport.java:77)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.processRequest(InitiationListener.java:91)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.access$000(InitiationListener.java:37)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener$1.run(InitiationListener.java:63)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:35:12,476 (DataTransferManager.java:415) failed to connect to eazycode@saros-con.imp.fu-berlin.de/Saros using transport: XMPP-Socks5-Transport
java.io.IOException: [SOCKS5Transport] waiting for a response session timed out (30000ms) and could not establish a connection from this side, too:
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishBinaryChannel(Socks5Transport.java:592)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport.connect(ByteStreamTransport.java:48)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.connectInternal(DataTransferManager.java:412)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.connect(DataTransferManager.java:297)
	at de.fu_berlin.inf.dpp.negotiation.IncomingSessionNegotiation.accept(IncomingSessionNegotiation.java:167)
	at de.fu_berlin.inf.dpp.ui.wizards.JoinSessionWizard$2.run(JoinSessionWizard.java:128)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smack.util.SyncPacketSend.getReply(SyncPacketSend.java:54)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:481)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:83)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishBinaryChannel(Socks5Transport.java:512)
	... 6 more
WARN  2016-12-21 20:35:12,682 (SubscriptionHandler.java:148) received error presence package from saros_233035903@conference.saros-con.imp.fu-berlin.de/xzer@saros-con.imp.fu-berlin.de/Saros, condition: not-authorized, message: A password is required to enter this room
WARN  2016-12-21 20:35:20,135 (Socks5Transport.java:411) [SOCKS5Transport] Couldn't accept request but still trying to establish a response connection: Could not establish socket with any provided host
ERROR 2016-12-21 20:35:21,528 (Socks5Transport.java:440) An error occurred while establishing a response connection 
item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smack.util.SyncPacketSend.getReply(SyncPacketSend.java:54)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:481)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:83)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishResponseSession(Socks5Transport.java:631)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport$1.call(Socks5Transport.java:108)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport$1.call(Socks5Transport.java:1)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
ERROR 2016-12-21 20:35:21,547 (ByteStreamTransport.java:105) could not establish bytestream connection to eazycode@saros-con.imp.fu-berlin.de/Saros
java.io.IOException: [SOCKS5Transport] Neither connection could be established. 
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.acceptNewRequest(Socks5Transport.java:446)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.acceptRequest(Socks5Transport.java:474)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport$1.incomingBytestreamRequest(ByteStreamTransport.java:77)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.processRequest(InitiationListener.java:91)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.access$000(InitiationListener.java:37)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener$1.run(InitiationListener.java:63)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
ERROR 2016-12-21 20:35:45,733 (DataTransferManager.java:270) failed to send TransferDescription [elementName=pong, namespace=de.fu_berlin.inf.dpp, recipient=eazycode@saros-con.imp.fu-berlin.de/Saros, sender=xzer@saros-con.imp.fu-berlin.de/Saros, compress=true], connection=[mode=IBB, id=saros-main-session] eazycode@saros-con.imp.fu-berlin.de/Saros:Stream is closed
java.io.IOException: Stream is closed
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession$IBBOutputStream.write(InBandBytestreamSession.java:634)
	at java.io.BufferedOutputStream.flushBuffer(Unknown Source)
	at java.io.BufferedOutputStream.flush(Unknown Source)
	at java.io.DataOutputStream.flush(Unknown Source)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.sendData(BinaryChannelConnection.java:525)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.splitAndSend(BinaryChannelConnection.java:560)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.send(BinaryChannelConnection.java:325)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendInternal(DataTransferManager.java:268)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendData(DataTransferManager.java:216)
	at de.fu_berlin.inf.dpp.net.internal.XMPPTransmitter.send(XMPPTransmitter.java:90)
	at de.fu_berlin.inf.dpp.project.internal.timeout.ClientSessionTimeoutHandler$2.run(ClientSessionTimeoutHandler.java:89)
	at de.fu_berlin.inf.dpp.util.ThreadUtils$1.run(ThreadUtils.java:38)
	at java.lang.Thread.run(Unknown Source)
ERROR 2016-12-21 20:35:45,754 (ClientSessionTimeoutHandler.java:93) failed to send pong
java.io.IOException: Stream is closed
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession$IBBOutputStream.write(InBandBytestreamSession.java:634)
	at java.io.BufferedOutputStream.flushBuffer(Unknown Source)
	at java.io.BufferedOutputStream.flush(Unknown Source)
	at java.io.DataOutputStream.flush(Unknown Source)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.sendData(BinaryChannelConnection.java:525)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.splitAndSend(BinaryChannelConnection.java:560)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.send(BinaryChannelConnection.java:325)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendInternal(DataTransferManager.java:268)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendData(DataTransferManager.java:216)
	at de.fu_berlin.inf.dpp.net.internal.XMPPTransmitter.send(XMPPTransmitter.java:90)
	at de.fu_berlin.inf.dpp.project.internal.timeout.ClientSessionTimeoutHandler$2.run(ClientSessionTimeoutHandler.java:89)
	at de.fu_berlin.inf.dpp.util.ThreadUtils$1.run(ThreadUtils.java:38)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:35:45,832 (SarosSession.java:655) failed to notify user eazycode@saros-con.imp.fu-berlin.de/Saros about local session stop
java.io.IOException: not connected to eazycode@saros-con.imp.fu-berlin.de/Saros [connection identifier=saros-main-session]
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendData(DataTransferManager.java:206)
	at de.fu_berlin.inf.dpp.net.internal.XMPPTransmitter.send(XMPPTransmitter.java:90)
	at de.fu_berlin.inf.dpp.project.internal.SarosSession.stop(SarosSession.java:651)
	at de.fu_berlin.inf.dpp.project.SarosSessionManager.stopSarosSession(SarosSessionManager.java:340)
	at de.fu_berlin.inf.dpp.project.internal.timeout.SessionTimeoutHandler$3.run(SessionTimeoutHandler.java:115)
	at de.fu_berlin.inf.dpp.util.ThreadUtils$1.run(ThreadUtils.java:38)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:36:56,572 (Socks5Transport.java:544) [SOCKS5Transport] could not establish a connection to eazycode@saros-con.imp.fu-berlin.de/Saros, remote side could not connect to any offered stream hosts: item-not-found(404) Could not establish socket with any provided host
ERROR 2016-12-21 20:36:57,513 (ByteStreamTransport.java:105) could not establish bytestream connection to eazycode@saros-con.imp.fu-berlin.de/Saros
Could not establish socket with any provided host: item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.cancelRequest(Socks5BytestreamRequest.java:277)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.accept(Socks5BytestreamRequest.java:247)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamRequest.accept(Socks5BytestreamRequest.java:34)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.handleResponse(Socks5Transport.java:315)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.acceptRequest(Socks5Transport.java:471)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport$1.incomingBytestreamRequest(ByteStreamTransport.java:77)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.processRequest(InitiationListener.java:91)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener.access$000(InitiationListener.java:37)
	at org.jivesoftware.smackx.bytestreams.socks5.InitiationListener$1.run(InitiationListener.java:63)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:37:26,582 (DataTransferManager.java:415) failed to connect to eazycode@saros-con.imp.fu-berlin.de/Saros using transport: XMPP-Socks5-Transport
java.io.IOException: [SOCKS5Transport] waiting for a response session timed out (30000ms) and could not establish a connection from this side, too:
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishBinaryChannel(Socks5Transport.java:592)
	at de.fu_berlin.inf.dpp.net.internal.ByteStreamTransport.connect(ByteStreamTransport.java:48)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.connectInternal(DataTransferManager.java:412)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.connect(DataTransferManager.java:297)
	at de.fu_berlin.inf.dpp.negotiation.IncomingSessionNegotiation.accept(IncomingSessionNegotiation.java:167)
	at de.fu_berlin.inf.dpp.ui.wizards.JoinSessionWizard$2.run(JoinSessionWizard.java:128)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: item-not-found(404) Could not establish socket with any provided host
	at org.jivesoftware.smack.util.SyncPacketSend.getReply(SyncPacketSend.java:54)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:481)
	at org.jivesoftware.smackx.bytestreams.socks5.Socks5BytestreamManager.establishSession(Socks5BytestreamManager.java:83)
	at de.fu_berlin.inf.dpp.net.internal.Socks5Transport.establishBinaryChannel(Socks5Transport.java:512)
	... 6 more
WARN  2016-12-21 20:37:26,765 (SubscriptionHandler.java:148) received error presence package from saros_1435994860@conference.saros-con.imp.fu-berlin.de/xzer@saros-con.imp.fu-berlin.de/Saros, condition: not-authorized, message: A password is required to enter this room
WARN  2016-12-21 20:37:36,760 (CancelProjectSharingHandler.java:74) Inv[unknown user]: Received invitation cancel message
ERROR 2016-12-21 20:38:05,787 (DataTransferManager.java:270) failed to send TransferDescription [elementName=pong, namespace=de.fu_berlin.inf.dpp, recipient=eazycode@saros-con.imp.fu-berlin.de/Saros, sender=xzer@saros-con.imp.fu-berlin.de/Saros, compress=true], connection=[mode=IBB, id=saros-main-session] eazycode@saros-con.imp.fu-berlin.de/Saros:Stream is closed
java.io.IOException: Stream is closed
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession$IBBOutputStream.write(InBandBytestreamSession.java:634)
	at java.io.BufferedOutputStream.flushBuffer(Unknown Source)
	at java.io.BufferedOutputStream.flush(Unknown Source)
	at java.io.DataOutputStream.flush(Unknown Source)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.sendData(BinaryChannelConnection.java:525)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.splitAndSend(BinaryChannelConnection.java:560)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.send(BinaryChannelConnection.java:325)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendInternal(DataTransferManager.java:268)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendData(DataTransferManager.java:216)
	at de.fu_berlin.inf.dpp.net.internal.XMPPTransmitter.send(XMPPTransmitter.java:90)
	at de.fu_berlin.inf.dpp.project.internal.timeout.ClientSessionTimeoutHandler$2.run(ClientSessionTimeoutHandler.java:89)
	at de.fu_berlin.inf.dpp.util.ThreadUtils$1.run(ThreadUtils.java:38)
	at java.lang.Thread.run(Unknown Source)
ERROR 2016-12-21 20:38:05,797 (ClientSessionTimeoutHandler.java:93) failed to send pong
java.io.IOException: Stream is closed
	at org.jivesoftware.smackx.bytestreams.ibb.InBandBytestreamSession$IBBOutputStream.write(InBandBytestreamSession.java:634)
	at java.io.BufferedOutputStream.flushBuffer(Unknown Source)
	at java.io.BufferedOutputStream.flush(Unknown Source)
	at java.io.DataOutputStream.flush(Unknown Source)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.sendData(BinaryChannelConnection.java:525)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.splitAndSend(BinaryChannelConnection.java:560)
	at de.fu_berlin.inf.dpp.net.internal.BinaryChannelConnection.send(BinaryChannelConnection.java:325)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendInternal(DataTransferManager.java:268)
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendData(DataTransferManager.java:216)
	at de.fu_berlin.inf.dpp.net.internal.XMPPTransmitter.send(XMPPTransmitter.java:90)
	at de.fu_berlin.inf.dpp.project.internal.timeout.ClientSessionTimeoutHandler$2.run(ClientSessionTimeoutHandler.java:89)
	at de.fu_berlin.inf.dpp.util.ThreadUtils$1.run(ThreadUtils.java:38)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:38:05,911 (SarosSession.java:655) failed to notify user eazycode@saros-con.imp.fu-berlin.de/Saros about local session stop
java.io.IOException: not connected to eazycode@saros-con.imp.fu-berlin.de/Saros [connection identifier=saros-main-session]
	at de.fu_berlin.inf.dpp.net.internal.DataTransferManager.sendData(DataTransferManager.java:206)
	at de.fu_berlin.inf.dpp.net.internal.XMPPTransmitter.send(XMPPTransmitter.java:90)
	at de.fu_berlin.inf.dpp.project.internal.SarosSession.stop(SarosSession.java:651)
	at de.fu_berlin.inf.dpp.project.SarosSessionManager.stopSarosSession(SarosSessionManager.java:340)
	at de.fu_berlin.inf.dpp.project.internal.timeout.SessionTimeoutHandler$3.run(SessionTimeoutHandler.java:115)
	at de.fu_berlin.inf.dpp.util.ThreadUtils$1.run(ThreadUtils.java:38)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:38:31,715 (SWTUtils.java:144) could not execute runnable de.fu_berlin.inf.dpp.ui.widgets.chat.ChatControl$5$4@6ac099d6, UI thread is not available
StackTrace:
	at de.fu_berlin.inf.dpp.ui.util.SWTUtils.runSafeSWTAsync(SWTUtils.java:145)
	at de.fu_berlin.inf.dpp.ui.widgets.chat.ChatControl$5.disconnected(ChatControl.java:301)
	at de.fu_berlin.inf.dpp.communication.chat.AbstractChat.notifyJIDDisconnected(AbstractChat.java:72)
	at de.fu_berlin.inf.dpp.communication.chat.single.SingleUserChat.setConnected(SingleUserChat.java:125)
	at de.fu_berlin.inf.dpp.communication.chat.single.SingleUserChatService$1.connectionStateChanged(SingleUserChatService.java:60)
	at de.fu_berlin.inf.dpp.net.xmpp.XMPPConnectionService.setConnectionState(XMPPConnectionService.java:405)
	at de.fu_berlin.inf.dpp.net.xmpp.XMPPConnectionService.disconnect(XMPPConnectionService.java:304)
	at de.fu_berlin.inf.dpp.communication.connection.ConnectionHandler.disconnectXMPPInternal(ConnectionHandler.java:225)
	at de.fu_berlin.inf.dpp.communication.connection.ConnectionHandler.disconnect(ConnectionHandler.java:211)
	at de.fu_berlin.inf.dpp.Saros$1.run(Saros.java:304)
	at de.fu_berlin.inf.dpp.util.ThreadUtils$1.run(ThreadUtils.java:38)
	at java.lang.Thread.run(Unknown Source)
WARN  2016-12-21 20:38:31,729 (SWTUtils.java:144) could not execute runnable de.fu_berlin.inf.dpp.ui.widgets.chat.ChatControl$5$4@74fc03ff, UI thread is not available
StackTrace:
	at de.fu_berlin.inf.dpp.ui.util.SWTUtils.runSafeSWTAsync(SWTUtils.java:145)
	at de.fu_berlin.inf.dpp.ui.widgets.chat.ChatControl$5.disconnected(ChatControl.java:301)
	at de.fu_berlin.inf.dpp.communication.chat.AbstractChat.notifyJIDDisconnected(AbstractChat.java:72)
	at de.fu_berlin.inf.dpp.communication.chat.single.SingleUserChat.setConnected(SingleUserChat.java:126)
	at de.fu_berlin.inf.dpp.communication.chat.single.SingleUserChatService$1.connectionStateChanged(SingleUserChatService.java:60)
	at de.fu_berlin.inf.dpp.net.xmpp.XMPPConnectionService.setConnectionState(XMPPConnectionService.java:405)
	at de.fu_berlin.inf.dpp.net.xmpp.XMPPConnectionService.disconnect(XMPPConnectionService.java:304)
	at de.fu_berlin.inf.dpp.communication.connection.ConnectionHandler.disconnectXMPPInternal(ConnectionHandler.java:225)
	at de.fu_berlin.inf.dpp.communication.connection.ConnectionHandler.disconnect(ConnectionHandler.java:211)
	at de.fu_berlin.inf.dpp.Saros$1.run(Saros.java:304)
	at de.fu_berlin.inf.dpp.util.ThreadUtils$1.run(ThreadUtils.java:38)
	at java.lang.Thread.run(Unknown Source)
